### PR TITLE
2FA mobile app auth: show operation no and date

### PR DESCRIPTION
**Pull request by chilek**
…ed 'developer' ;-)

### CODE REVIEW

**Comment by chilek** [comment]
Added operation no and date presentaiton hopefully for both: sms and mobile app. Operation date has a little bit different format than this from sms.

**Comment by jwilk** [comment]
Can you give an example?
For SMS auth the date format is `2019-09-16T00:00:00`.

**Comment by chilek** [comment]
2019-09-16T12:57:15.393

I don't know why the current parse_timestamp() (from master/HEAD) didn't want to parse date properly...

**Comment by chilek** [comment]
Ok - I've restored previous regexp in parse_timestamp() as it appears to work.

**Comment by chilek** [comment]
But this change https://github.com/jwilk/mbank-cli/pull/77/files#diff-9f77a56380d90ae7dc01803fd61f3df4R768 seems to be mandatory.

**Comment by chilek** [comment]
What was the purpose of:
```perl
$discard_time or ($time =~ m/(\A)?[0:.]+(\z)?/)
```
?

File: mbank-cli [hunk 1]
```diff
@@ -1225,8 +1225,14 @@ sub do_login  ## no critic (ExcessComplexity)
         $error_code eq ''
             or scraping_error("login.2fa.init.error: $error_code");
         my $auth_mode = $data->{AuthMode} // '';
+        my $operation_date = $data->{OperationDate} // '';
+        $operation_date = parse_timestamp($operation_date)
```
Comment (jwilk) [inline mbank-cli]:
```suggestion
        $operation_date = parse_timestamp($operation_date, discard_time => 1)
```
and then you won't need to fiddle with the regexp.
Without `discard_time`, the function checks that the time is all zeros. (Yes, this is all awful.)

Comment (chilek) [reply mbank-cli]: https://github.com/jwilk/mbank-cli/pull/77/commits/992f258de7e570c5fd6b0b8f7f4fc36f313daa88

**Comment by jwilk** [comment]
I took the liberty of squashing commits and editing commit messages in order to satisfy the pedant in me.
Commited as 3840bb3de2992e2b85a7614d61decc61fc9975bb.

**Comment by chilek** [comment]
Yeah I like squashing too ;-)

**Pull request closed without merging**